### PR TITLE
Added empty step validation

### PR DESCRIPTION
**Pull request by siddhartha97**
**What type of PR is this?**

**What this PR does / why we need it**:
/kind feature

**Which issue(s) this PR fixes**:
Validates more aspect of CL2 config

Part of #1746 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by siddhartha97** [comment]
signed cla

**Comment by siddhartha97** [comment]
> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please follow instructions at https://git.k8s.io/community/CLA.md#the-contributor-license-agreement to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment and we'll verify. Thanks.
> 
> * If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address.  Check your existing CLA data and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * If you signed the CLA as a corporation, please sign in with your organization's credentials at https://identity.linuxfoundation.org/projects/cncf to be authorized.
> * If you have done the above and are still having issues with the CLA being reported as unsigned, please log a ticket with the Linux Foundation Helpdesk: https://support.linuxfoundation.org/
> * Should you encounter any issues with the Linux Foundation Helpdesk, send a message to the backup e-mail support address at: [login-issues@jira.linuxfoundation.org](mailto:login-issues@jira.linuxfoundation.org)

CLA signed (:

**Comment by siddhartha97** [comment]
CLA signed! (:

File: clusterloader2/api/validation.go [hunk 1]
```diff
@@ -85,6 +86,11 @@ func (v *ConfigValidator) validateNamespace(ns *NamespaceConfig, fldPath *field.
 
 func (v *ConfigValidator) validateStep(s *Step, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
+	if s == nil || reflect.DeepEqual(*s, Step{}) {
```
[the remark this answers is not in the document]
Comment (siddhartha97) [reply clusterloader2/api/validation.go:89]: will add the tests! by one level up do you mean somewhere here, https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/api/validation.go#L59

[the remark this answers is not in the document]
Comment (siddhartha97) [reply clusterloader2/api/validation.go:89]: will add the tests! By one level up do you mean somewhere here? https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/api/validation.go#L59

[the remark this answers is not in the document]
Comment (lyzs90) [reply clusterloader2/api/validation.go:89]: Sorry my link looks broken. Yes before the loop.

**Comment by marseel** [comment]
/close
We already have empty steps validation.

**Pull request closed without merging**